pkg/scheduler/scheduler.go

```diff
@@ -467,7 +476,7 @@ func (sched *Scheduler) addAllEventHandlers() {
 					klog.ErrorDepth(5, fmt.Sprintf("failed to parse labelSelector in Subscription %s: %v", key, err))
 					continue
 				}
-				if !selector.Matches(labels.Set(mcls.Labels)) {
+				if !selector.Matches(labels.Set(mcls.Labels)) && !utils.ContainsString(sched.bindingClusters[key], klog.KObj(mcls).String()) {
```

Comment:
IMO `enqueueSubscriptionForClusterFunc` is the only place for this change. We can take two params `oldMcls` and `newMcls` for this function.